js/model3D/shared/Canvas3D.svelte

```diff
@@ -52,176 +42,91 @@
-
-		mounted = true;
-
 		return () => {
-			scene.dispose();
-			engine.dispose();
-			window.removeEventListener("resize", onWindowResize);
+			viewer.dispose();
 		};
 	});
 
 	$: mounted && load_model(resolved_url);
 
-	function load_model(url: string | undefined): void {
-		if (scene) {
-			// Dispose of the previous model before loading a new one
-			scene.meshes.forEach((mesh) => {
-				mesh.dispose();
-			});
-
-			if (point_cloud_system) {
-				point_cloud_system.dispose();
-				point_cloud_system = null;
-			}
+	function setRenderingMode(pointsCloud: boolean, wireframe: boolean): void {
+		viewerDetails.scene.forcePointsCloud = pointsCloud;
+		viewerDetails.scene.forceWireframe = wireframe;
+	}
 
-			// Load the new model
-			if (url) {
-				BABYLON.SceneLoader.ShowLoadingScreen = false;
-				BABYLON.SceneLoader.Append(
-					url,
-					"",
-					scene,
-					() => {
-						if (display_mode === "point_cloud") {
-							create_point_cloud(scene);
-						} else if (display_mode === "wireframe") {
-							create_wireframe(scene);
-						} else {
-							create_camera(scene, camera_position, zoom_speed, pan_speed);
+	function load_model(url: string | undefined): void {
+		if (viewer) {
+			viewer
+				.loadModel(url!, {
+					pluginOptions: {
+						obj: {
+							importVertexColors: true
 						}
-					},
-					undefined,
-					undefined,
-					"." + value.path.split(".").pop()
-				);
-			}
+					}
+				})
+				.then(() => {
+					if (display_mode === "point_cloud") {
+						setRenderingMode(true, false);
+					} else if (display_mode === "wireframe") {
+						setRenderingMode(false, true);
+					} else {
+						update_camera(camera_position, zoom_speed, pan_speed);
+					}
+				});
 		}
 	}
 
-	function create_camera(
-		scene: BABYLON.Scene,
+	function update_camera(
 		camera_position: [number | null, number | null, number | null],
 		zoom_speed: number,
 		pan_speed: number
 	): void {
-		scene.createDefaultCamera(true, true, true);
-		var helperCamera = scene.activeCamera! as BABYLON.ArcRotateCamera;
+		const camera = viewerDetails.camera;
 		if (camera_position[0] !== null) {
-			helperCamera.alpha = BABYLON.Tools.ToRadians(camera_position[0]);
+			camera.alpha = BABYLON.Tools.ToRadians(camera_position[0]);
 		}
 		if (camera_position[1] !== null) {
-			helperCamera.beta = BABYLON.Tools.ToRadians(camera_position[1]);
+			camera.beta = BABYLON.Tools.ToRadians(camera_position[1]);
 		}
 		if (camera_position[2] !== null) {
-			helperCamera.radius = camera_position[2];
+			camera.radius = camera_position[2];
 		}
-		helperCamera.lowerRadiusLimit = 0.1;
+		camera.lowerRadiusLimit = 0.1;
 		const updateCameraSensibility = (): void => {
-			helperCamera.wheelPrecision = 250 / (helperCamera.radius * zoom_speed);
-			helperCamera.panningSensibility =
-				(10000 * pan_speed) / helperCamera.radius;
+			camera.wheelPrecision = 250 / (camera.radius * zoom_speed);
+			camera.panningSensibility = (10000 * pan_speed) / camera.radius;
```

Comment:
This might not be needed as the Babylon Viewer already adjusts panning sensitivity and speed based on the radius: https://github.com/BabylonJS/Babylon.js/blob/5d62ed08213d5421dc8f423358bb90c7f5ddae10/packages/tools/viewer/src/viewer.ts#L1589-L1591